OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectIcon.cs

```diff
@@ -0,0 +1,142 @@
+
+                // TODO confirm BYOND behavior of invalid args for icon, dir, and frame
+
+                if (state is not null && state.Value.TryGetValueAsString(out var iconState))
+                {
+                    State = iconState;
+                }
+                else
+                {
+                    State = null;
+                }
+
+                if (dir is not null && dir.Value.TryGetValueAsInteger(out var dirVal) && (AtomDirection)dirVal != AtomDirection.None)
+                {
+                    Direction = (AtomDirection)dirVal;
+                }
+                else
+                {
+                    Direction = null;
+                }
+
+                if (frame is not null && frame.Value.TryGetValueAsInteger(out var frameVal))
+                {
+                    //TODO: Figure out how many frames an icon can have and see if this needs to be bigger than a byte
+                    Frame = Convert.ToByte(frameVal - 1); //1-indexed
+                }
+                else
+                {
+                    Frame = null;
+                }
+
+                if (moving is not null && moving != DreamValue.Null)
+                {
+                    if (moving.Value.TryGetValueAsInteger(out var movingVal) && movingVal == 0)
+                    {
+                        Moving = DreamIconMovingMode.NonMovement;
+                    }
+                    else
+                    {
+                        Moving = DreamIconMovingMode.Movement;
+                    }
+                }
+                else
+                {
+                    Moving = DreamIconMovingMode.Both;
+                }
+            }
+        }
+
+        public override void OnObjectCreated(DreamObject dreamObject, DreamProcArguments creationArguments) {
+            base.OnObjectCreated(dreamObject, creationArguments);
+
+            DreamValue icon = creationArguments.GetArgument(0, "icon");
+            DreamValue state = creationArguments.GetArgument(1, "icon_state");
+            DreamValue dir = creationArguments.GetArgument(2, "dir");
+            DreamValue frame = creationArguments.GetArgument(3, "frame");
+            DreamValue moving = creationArguments.GetArgument(4, "moving");
+
+            DreamIconObject dreamIconObject;
+
+            if (icon.TryGetValueAsDreamObjectOfType(DreamPath.Icon, out DreamObject copyFrom)) {
+                dreamIconObject = ObjectToDreamIcon[copyFrom];
+            } else if (icon.TryGetValueAsString(out string fileString))
+            {
+                var ext = Path.GetExtension(fileString);
+                switch (ext) // TODO implement other icon file types
+                {
+                    case ".dmi":
+                        dreamIconObject = new DreamIconObject(_rscMan.LoadResource(fileString), state, dir, frame, moving);
+                        dreamObject.SetVariableValue("icon", new DreamValue(fileString));
```

Comment:
`/icon.icon` isn't the icon path. It seems to be some kind of internal structure, `usr << I.icon` leads to a "bad output value" error.
```suggestion
```